squadutilslib.py

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# vim: set ts=4
+#
+# Copyright 2023-present Linaro Limited
+#
+# SPDX-License-Identifier: MIT
+
+
+import logging
+import os
+import re
+from pathlib import Path
+
+import requests
+from requests import HTTPError
+from squad_client.core.models import ALL, Build, Squad
+from squad_client.utils import first, getid
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ReproducerNotFound(Exception):
+    """
+    Raised when no reproducer can be found.
+    """
+
+    def __init__(self, message="No reproducer found"):
+        super().__init__(message)
+
+
+def get_file(path, filename=None):
+    """
+    Download file if a URL is passed in, then return the filename of the
+    downloaded file. If an existing file path is passed in, return the path. If
+    a non-existent path is passed in, raise an exception.
+    """
+    logger.info(f"Getting file from {path}")
+    if re.search(r"https?://", path):
+        request = requests.get(path, allow_redirects=True)
+        request.raise_for_status()
+        if not filename:
+            filename = path.split("/")[-1]
+        else:
+            output_file = Path(filename)
+            output_file.parent.mkdir(exist_ok=True, parents=True)
+
+        with open(filename, "wb") as f:
+            f.write(request.content)
+        return filename
+    elif os.path.exists(path):
+        return path
+    else:
+        raise Exception(f"Path {path} not found")
+
+
+def download(
```

Comment:
I probably don't need to create a wrapper here, but I created my own adapted version of download_tests because it was not possible to create a filter for the build name.

Reply:
As we talked about not sure we need this.